lib/posix/pthread.c

```diff
@@ -222,15 +222,16 @@ static bool is_posix_policy_prio_valid(uint32_t priority, int policy)
 		return true;
 	}
 
-	LOG_ERR("Invalid piority %d and / or policy %d", priority, policy);
+	LOG_ERR("Invalid priority %d and / or policy %d", priority, policy);
 
 	return false;
 }
 
-static uint32_t zephyr_to_posix_priority(int32_t z_prio, int *policy)
+/* Non-static so that they can be tested in ztest */
+int zephyr_to_posix_priority(int z_prio, int *policy)
```

Comment:
POSIX priorities are only >= 0.

I guess it would be fine to return a negative value to signal an error. I thought there were assertions to that effect though.